### ISSUE TITLE
Do we have an update for Swift 4.x?

### ISSUE THREAD

**Issue by kuangzq** [emqtt/CocoaMQTT#228]
As the title.

**Comment by sunil-targe** [comment]
We're also facing issue with Xcode 10 and Swift 4.2. ;(

**Comment by andy1247008998** [comment]
This issue is caused by pod SwiftTimer.

[SwiftyTimer.swift](https://github.com/radex/SwiftyTimer/blob/master/Sources/SwiftyTimer.swift)
change Line 107-108 to:
```
    public func start(runLoop: RunLoop = .current, modes: RunLoop.Mode...) {
        let modes = modes.isEmpty ? [RunLoop.Mode.default] : modes
```
The real problem is: This pod is no longer actively maintained, we may need a replacement.

**Comment by Radagan** [comment]
Specifically, Swift 4.x where x = 2...

**Comment by asowers1** [comment]
I added a PR to SwiftyTimer for Swift 4.2 source compatibility. Go +1 it for support. https://github.com/radex/SwiftyTimer/pull/45

**Comment by asowers1** [comment]
Would it be acceptable for me to submit a PR to CocoaMQTT embedding the source of SwiftyTimer if Radex isn't able to respond in a timely manner?

**Comment by HJianBo** [comment]
I released version 1.1.3. That embedded the SwiftyTimer code and remove it from dependency.
If there are still any problems, please let me know.

**Comment by asowers1** [comment]
I believe this issue can be closed.
